client/my-sites/comments/comment/index.jsx

```diff
@@ -44,7 +53,40 @@ export class Comment extends Component {
 
 	componentWillReceiveProps( nextProps ) {
 		const { isBulkMode: wasBulkMode } = this.props;
-		const { isBulkMode } = nextProps;
+		const {
+			commentId,
+			hasScrolledToComment,
+			isBulkMode,
+			isLoadedAll,
+			isLoading,
+			onCommentLoad,
+			onScrollToComment,
+		} = nextProps;
+
+		if ( ! isLoading ) {
+			onCommentLoad( commentId );
+		}
+
+		if (
+			! hasScrolledToComment &&
+			isLoadedAll &&
+			!! window &&
+			`#comment-${ commentId }` === window.location.hash
+		) {
+			const commentNode = ReactDom.findDOMNode( this.commentCard );
+			const commentOffsetTop = commentNode.offsetTop - COMMENT_SCROLL_TOP_MARGIN;
+			scrollTo( {
+				x: 0,
+				y: commentOffsetTop,
+				duration: 1,
+				onComplete: () => {
+					if ( commentOffsetTop !== window.scrollY ) {
+						window.scrollTo( 0, commentOffsetTop );
+					}
+					onScrollToComment();
```

Comment:
I believe this should still moved above the scrollTo

Reply:
Casually testing, we're triggering the scroll 4 times on single page render! The `isScrolling` bit will prevent us from calling `ScrollTo` those additional times. eg `! hasScrolledToComment && ! isScrolling`

<img width="194" alt="screen shot 2017-12-07 at 12 08 42 pm" src="https://user-images.githubusercontent.com/1270189/33736476-9729a010-db47-11e7-8df4-8aea08b42674.png">